### ISSUE TITLE
PettingZoo simple_spread example doesnt learn

### ISSUE THREAD

**Issue by mlanas** [instadeepai/Mava#315]
# Problem

I'm running the PettingZoo simple_spread Mava example ([run_maddpg.py](https://github.com/mlanas/Mava/blob/develop/examples/petting_zoo/mpe/simple_spread/feedforward/decentralised/run_maddpg.py))  from the [develop](https://github.com/instadeepai/Mava/tree/develop)  branch and the MeanEpisodeReturn does not improve.

![image](https://user-images.githubusercontent.com/56938299/133464124-2273ada9-7cd2-4d91-88e2-976ebf8e727a.png)

Is this the expected behaviour? Or should I maybe let it train longer?

# Execution

```bash
make run-tensorboard EXAMPLE=examples/petting_zoo/mpe/simple_spread/feedforward/decentralised/run_maddpg.py
```

**Comment by KaleabTessera** [comment]
Hi @mlanas :wave: 

So maddpg usually takes quite a while to train on PZ simple spread. We trained it for ~3 000 000 executor steps / 24h to get to ~-25 return. Mad4pg does better and gets close to -20. 

Here is the mad4pg config: 


Config | Param
-- | --
batch_size | 1024
critic_optimizer | Adam
critic_optimizer_lr | 0.0001
discount | 0.99
executor_variable_update_period | 100
max_gradient_norm | None
policy_optimizer | Adam
policy_optimizer_lr | 0.0001
sigma | 0.3
target_averaging | False

Hope this answers your question. :)

**Comment by mlanas** [comment]
Hey @KaleabTessera ! Thanks for the answer, looks like I did not train long enough 😅. Do increasing the number of executors speeds up the training? And do you know why is there such a big difference with the debugging environment? They are really similar, but the debug env trains in 5 min with maddpg.

**Comment by KaleabTessera** [comment]
@mlanas Yes, increasing the number of executors should speed up training, but it can also change the training trajectory since the data gets added to our dataset faster. 

@DriesSmit Maybe you can answer the exact differences between our debug simple spread and PZ simple spread?

**Comment by DriesSmit** [comment]
Hey @mlanas. The debugging simple spread was made to be much easier than PZ simple spread one. In the debugging environment, the agents have specific landmarks that each agent has to go to which makes it easier to learn. In the PZ simple spread, they don't have specific landmarks they need to go to which makes the coordination problem a bit more difficult. The rewards also have less variance than the PZ equivalent.

**Comment by mlanas** [comment]
Thanks for the help guys, I manage to make it work!

**Comment by mlanas** [comment]
Hey @KaleabTessera do you happen to have a working config for other MPE envs like tag or adversary? It would really help me if you can share it with me, thanks in advance :)

**Comment by KaleabTessera** [comment]
Hi @mlanas , for `Simple Speaker-listener` , we used the hyperparams above, but we used a policy and critic network size of [64,64], a max gradient norm of 1, a discount of 0.95 and a policy_optimizer_lr of 1e-2. This followed from the maddpg codebase - https://github.com/openai/maddpg/blob/3ceefa0ada3ff31d633dd0bde8ff95213ce99be3/experiments/train.py on mpe.

**Comment by mlanas** [comment]
Thank you very much @KaleabTessera
